internals/overlord/state/state.go

```
@@ -243,12 +197,19 @@ func (s *State) UnmarshalJSON(data []byte) error {
 		return err
 	}
 	s.data = unmarshalled.Data
+
+	// Load legacy identities if new identities are not in Get/Set data.
+	if !s.data.has("identities") && len(unmarshalled.LegacyIdentities) > 0 {
```

It seems like we shouldn't be able to easily get into a state where we have both the new `data["identities"]` and the old top-level key, and this conditional encodes that assumption by silently ignoring `LegacyIdentities` if we already have `data["identities"]`. Is that the best approach, or would it be worth emitting a warning about a bad database migration if we end up with both?